File: 4033-longest-subsequence-with-non-zero-bitwise-xor/longest-subsequence-with-non-zero-bitwise-xor.java
class Solution {
    public int longestSubsequence(int[] nums) {
        int i,j,x=0;
        int n = nums.length;
        boolean nonzero = false;
        for(i=0;i<n;i++){
            x^=nums[i];
            if(nums[i]!=0)nonzero=true;
        }
        if(x!=0)return n;
        if(nonzero)return n-1;
        return 0;
    }
}

File: 4033-longest-subsequence-with-non-zero-bitwise-xor/longest-subsequence-with-non-zero-bitwise-xor.cpp
class Solution {
public:
    int longestSubsequence(vector<int>& nums) {
        int i,j;
        int xori=0;
        bool nonzero = false;
        for(auto x : nums){
            xori^=x;
            if(x!=0)nonzero=true;
        }
        if(xori!=0)return nums.size();
        if(nonzero)return nums.size()-1;
        return 0;
    }
};